### DISCUSSION TITLE
Do the screenshots generated by this extension almost always match the ones generated by Firefox's devtools?

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [M-Reimer/savescreenshot#64] [Q&A]
I've been mostly using a different screenshot extension for quite some time.  I've always liked it quite a bit, but recently I've noticed that more and more screenshots of long web pages don't look quite right.  This is often due to fixed-positioned HTML elements.

When I use Firefox's devtool's integrated screenshot tool, those pages always are captured correctly.  I don't have sufficient examples to test at the moment, so I was wondering if this tool will likely have the same success as Firefox's devtool's integrated screenshot tool, or if it will likely have the same limitations as other popular browser extensions?

**Comment by M-Reimer** [comment]
Firefox extends the API, used for capturing the image inside a tab, to allow for overlong screenshots.

Many screenshot Add-ons are "cross browser". As this API extension is "Firefox only", those Add-ons often don't use it but manually scroll the page to capture multiple times and then merge many images.

As I don't care about Chrome, I can make use of the Firefox built in features and yes, it is possible that this is basically the same interface as in dev tools.

**Reply by Gitoffthelawn** [reply]
Sounds great. Thanks.  Has this Firefox API extension been around for years, or is it recent?

**Reply by M-Reimer** [reply]
Pretty recent. It was added when multiprocess was implemented.

**Reply by Gitoffthelawn** [reply]
Believe it or not, that was 7 years ago! :)

**Reply by M-Reimer** [reply]
Interesting. For some time this wasn't really relevant and there was a "unofficial" API that I kept using as long as it lasted. This is where I updated my Add-on: https://github.com/M-Reimer/savescreenshot/commit/a7727963059238a184af4da81f2723a2d8c80703

**Reply by Gitoffthelawn** [reply]
Thanks.  I see from the commit, it was for Firefox v82.  That's close enough for what I wanted to know.

BTW, that "unofficial" API is now gone from Firefox, correct?

**Reply by M-Reimer** [reply]
It is not gone but has issues. In some cases embedded frames are just blank in screenshots.
